pinot-core/src/main/java/org/apache/pinot/core/operator/filter/AndFilterOperator.java

```diff
@@ -52,8 +53,21 @@ public AndFilterOperator(List<BaseFilterOperator> filterOperators, @Nullable Map
   protected BlockDocIdSet getTrues() {
     Tracing.activeRecording().setNumChildren(_filterOperators.size());
     List<BlockDocIdSet> blockDocIdSets = new ArrayList<>(_filterOperators.size());
+    long totalEntriesScanned = 0L;
     for (BaseFilterOperator filterOperator : _filterOperators) {
-      blockDocIdSets.add(filterOperator.getTrues());
+      BlockDocIdSet blockDocIdSet = filterOperator.getTrues();
+      BlockDocIdSet optimizedDocIdSet = blockDocIdSet.getOptimizedDocIdSet();
```

Comment:
Entries scanned can be reported incorrectly here. If `blockDocIdSet` is a `ShortCircuitingDocIdSet`, the getOptimizedDocIdSet call will return an empty doc id set, and you will miss out on adding the entries scanned of the nested doc id set.

I think it's better to add entries scanned in filter to the existing EmptyDocIdSet. The default instance of EmptyDocIdSet could still be set to zero entriesScannedInFilter.